File: AndroidMaterialBottomBar/app/src/main/java/com/example/shakil/androidmaterialbottombar/MainActivity.java
package com.example.shakil.androidmaterialbottombar;

import android.support.design.bottomappbar.BottomAppBar;
import android.support.design.widget.FloatingActionButton;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    BottomAppBar bottomAppBar;
    FloatingActionButton fab;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        fab = findViewById(R.id.fab);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(MainActivity.this, "Clicked on FAB", Toast.LENGTH_SHORT).show();
            }
        });
        bottomAppBar = findViewById(R.id.bottom_app_bar);
        bottomAppBar.replaceMenu(R.menu.main_menu);
        bottomAppBar.setOnMenuItemClickListener(new Toolbar.OnMenuItemClickListener() {
            @Override
            public boolean onMenuItemClick(MenuItem menuItem) {
                switch (menuItem.getItemId()){
                    case R.id.action_cart:
                        Toast.makeText(MainActivity.this, "Clicked on cart", Toast.LENGTH_SHORT).show();
                        break;

                    case R.id.action_shipping:
                        Toast.makeText(MainActivity.this, "Clicked on shipping", Toast.LENGTH_SHORT).show();
                        break;

                    case R.id.action_customer:
                        Toast.makeText(MainActivity.this, "Clicked on customer", Toast.LENGTH_SHORT).show();
                        break;
                }
                return true;
            }
        });
    }
}

File: KotlinMaterialBottomBar/app/src/main/java/com/example/shakil/kotlinmaterialbottombar/MainActivity.kt
package com.example.shakil.kotlinmaterialbottombar

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        fab.setOnClickListener {
            Toast.makeText(this@MainActivity, "FAB click", Toast.LENGTH_SHORT).show()
        }

        bottom_app_bar.replaceMenu(R.menu.menu_main)

        bottom_app_bar.setOnMenuItemClickListener { menuItem ->
            when (menuItem.itemId) {
                R.id.action_cart -> Toast.makeText(this@MainActivity, "Cart click", Toast.LENGTH_SHORT).show()
                R.id.action_shipping -> Toast.makeText(this@MainActivity, "Shipping click", Toast.LENGTH_SHORT).show()
                R.id.action_customer -> Toast.makeText(this@MainActivity, "Customer click", Toast.LENGTH_SHORT).show()
            }
            true
        }
    }
}
